plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.test.ts

```diff
@@ -542,4 +671,368 @@ describe('BitbucketServerEntityProvider', () => {
+          },
+        ],
+      },
+    });
+    const providers = BitbucketServerEntityProvider.fromConfig(config, {
+      logger,
+      events,
+      schedule,
+    });
+
+    expect(providers).toHaveLength(1);
+    expect(providers[0].getProviderName()).toEqual(
+      'bitbucketServer-provider:default',
+    );
+
+    const locationEntities = [
+      {
+        apiVersion: 'backstage.io/v1alpha1',
+        kind: 'Location',
+        metadata: {
+          annotations: {
+            'backstage.io/managed-by-location': `url:https://${host}/projects/project-test/repos/repo-test/browse/catalog-info.yaml`,
+            'backstage.io/managed-by-origin-location': `url:https://${host}/projects/project-test/repos/repo-test/browse/catalog-info.yaml`,
+            [`${host}/repo-url`]: `https://${host}/projects/project-test/repos/repo-test/browse/catalog-info.yaml`,
+            'bitbucket.org/default-branch': 'master',
+          },
+          name: 'generated-77f4323822420990f8c3e3c981d38c2dec4ae3a6',
+        },
+        spec: {
+          presence: 'optional',
+          target: `https://${host}/projects/project-test/repos/repo-test/browse/catalog-info.yaml`,
+          type: 'url',
+        },
+      },
+      {
+        apiVersion: 'backstage.io/v1alpha1',
+        kind: 'Location',
+        metadata: {
+          annotations: {
+            'backstage.io/managed-by-location': `url:https://${host}/projects/other-project/repos/other-repo/browse/catalog-info.yaml`,
+            'backstage.io/managed-by-origin-location': `url:https://${host}/projects/other-project/repos/other-repo/browse/catalog-info.yaml`,
+            [`${host}/repo-url`]: `https://${host}/projects/other-project/repos/other-repo/browse/catalog-info.yaml`,
+            'bitbucket.org/default-branch': 'master',
+          },
+          name: 'generated-d8d4944c30c2906dfee172ddda9537f9893b2c0f',
+        },
+        spec: {
+          presence: 'optional',
+          target: `https://${host}/projects/other-project/repos/other-repo/browse/catalog-info.yaml`,
+          type: 'url',
+        },
+      },
+    ];
+
+    const deferredEntities = toDeferredEntities(
+      locationEntities,
+      providers[0].getProviderName(),
+    );
+
+    expect(deferredEntities).toEqual([
+      {
+        locationKey: providers[0].getProviderName(),
+        entity: locationEntities[0],
+      },
+      {
+        locationKey: providers[0].getProviderName(),
+        entity: locationEntities[1],
+      },
+    ]);
+  });
+
+  it('refresh onRepoPush', async () => {
+    const schedule = new PersistingTaskRunner();
+    const keptModule = createLocationEntity(
+      test1RepoUrl,
+      `/kept-module:${targetPath}`,
+      'master',
+    );
+    const entityProviderConnection: EntityProviderConnection = {
+      applyMutation: jest.fn(),
+      refresh: jest.fn(),
+    };
+
+    setupRepositoryReqHandler('master');
+
+    authService.getPluginRequestToken.mockResolvedValue({
```

Comment:
when you switch to the CatalogService which accepts a credentials object instead of tokens, you'll notice that the tests are afffected as well in a positive way - the mock auth service already has a set of mock credentials that it uses automatically, so you can check that you were called with `mockCredentials.service()` etc if you like ([example](https://github.com/backstage/backstage/blob/f1ac148efdb7d23319e1ab90154dda43e5a4e0c3/plugins/permission-node/src/ServerPermissionClient.test.ts#L107))